workspaces/src/network/result.rs

```diff
@@ -81,6 +81,22 @@ impl CallExecutionDetails {
         };
         base64::decode(result).map_err(Into::into)
     }
+
+    /// Convert the execution details into a Result if its status is not a successful one.
+    /// Useful for checking if the call was successful and forwarding the error upwards.
+    fn try_into_result(self) -> anyhow::Result<Self> {
+        match self.status {
+            FinalExecutionStatus::Failure(ref err) => anyhow::bail!(err.clone()),
+            FinalExecutionStatus::NotStarted => anyhow::bail!("Transaction not started."),
+            FinalExecutionStatus::Started => anyhow::bail!("Transaction still being processed."),
```

Comment:
noting that a user might want to use the details of the call execution when handling the error, even if it's downcasting to the type.

Fine as is for now though I think

Reply:
yeah we should make a custom Error to capture some of the details like logs later when the other refactor/results PR lands